### DISCUSSION TITLE
Calculate subtitle hash for remote files

### DISCUSSION THREAD

**Discussion by Suraj-Yadav** [tomasklaen/uosc#939] [Ideas]
Can we extend the hash calculation logic to work for remote URLs? 

I was thinking of using https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests, to fetch the required chunks without going through the whole file.

I have made some rough changes which are working, so I can open a PR if this can be a useful feature.

**Comment by tomasklaen** [comment]
But this wouldn't work for 99% of url usage in mpv, which goes through yt-dlp. We'd have to somehow replicate how mpv fetches the files from urls (including user configured yt-dlp quality settings) to ensure we're requesting same files. And the value of doing that seems non existent, since how many url hosted videos/streams will have subtitles on opensubtitles attached to them?

This might only be useful if someone is watching some scene releases hosted on a remote server, which is such a niche use case... I don't want to add and maintain a code for that. Especially since I plan on eventually rewriting ziggy in something else to get rid of the antivirus false positives, and have smaller binaries.

**Comment by Suraj-Yadav** [comment]
I agree that this is a very niche use case.
